policy-service/src/main/java/uk/gov/gchq/palisade/service/policy/PolicyApplication.java

```diff
@@ -16,26 +16,133 @@
+    private final ResourceConfiguration resourceConfig;
+
+    /**
+     * Autowire Akka objects in constructor for application ready event
+     *
+     * @param runners        collection of all Akka {@link RunnableGraph}s discovered for the application
+     * @param materializer   the Akka {@link Materializer} configured to be used
+     * @param service        specifically policyServiceCachingProxy used for pre-population
+     * @param policyConfig   resourceConfig used to create the policy object used in pre-population
+     * @param userConfig     resourceConfig used to create the user object used in pre-population
+     * @param resourceConfig resourceConfig used to create the resource object used in pre-population
+     * @param executor       an executor for any {@link CompletableFuture}s (preferably the application task executor)
+     */
+    public PolicyApplication(
+            final Collection<RunnableGraph<?>> runners,
+            final Materializer materializer,
+            final PolicyServiceCachingProxy service,
+            final PolicyConfiguration policyConfig,
+            final UserConfiguration userConfig,
+            final ResourceConfiguration resourceConfig,
+            @Qualifier("threadPoolTaskExecutor") final Executor executor) {
+        this.runners = new HashSet<>(runners);
+        this.materializer = materializer;
+        this.service = service;
+        this.policyConfig = policyConfig;
+        this.userConfig = userConfig;
+        this.resourceConfig = resourceConfig;
+        this.executor = executor;
+    }
+
+    /**
+     * Application entrypoint, creates and runs a spring application, passing in the given command-line args
+     *
+     * @param args command-line arguments passed to the application
+     */
     public static void main(final String[] args) {
-        LOGGER.debug("PolicyApplication started with: {} {} {}", PolicyApplication.class.toString(), "main", Arrays.toString(args));
+        LOGGER.debug("PolicyApplication started with: {}", (Object) args);
         new SpringApplicationBuilder(PolicyApplication.class).web(WebApplicationType.SERVLET)
                 .run(args);
     }
 
+    /**
+     * Init post construct.
+     * Pre-population profile used in the KafkaContractTests for pre-populating the cache.
+     * Example profile used for pre-populating the cache for the example
+     */
+    @EventListener(ApplicationReadyEvent.class)
+    @Profile({"pre-population, example"})
+    public void initPostConstruct() {
+        // Add example Policies to the policy-service cache
+        LOGGER.debug("Pre-populating using policy config: {}", policyConfig.getClass());
+        LOGGER.debug("Pre-populating using user config: {}", userConfig.getClass());
+        LOGGER.debug("Pre-populating using resource config: {}", resourceConfig.getClass());
+        policyConfig.getPolicies().stream()
+                .peek(e -> LOGGER.info("pre-pop stream {}", e))
```

Comment:
This could be a more useful message

Reply:
.peek is generally not needed, so removed it as other logging is in place